components/brave_vpn/brave_vpn_service.cc

```diff
@@ -882,28 +884,37 @@ void BraveVpnService::OnResume() {}
 void BraveVpnService::GetPurchaseToken(GetPurchaseTokenCallback callback) {
   std::string purchase_token_string = "";
   std::string package_string = "com.brave.browser";
+  std::string product_id_string = "brave-firewall-vpn-premium";
 
   // Get the Android purchase token (for Google Play Store).
   // The value for this is validated on the account.brave.com side
   auto* purchase_token =
-      local_prefs_->FindPreference(prefs::kBraveVPNPurchaseTokenAndroid);
+      profile_prefs_->FindPreference(prefs::kBraveVPNPurchaseTokenAndroid);
```

Comment:
we don't need to copy previous value from local state?

Reply:
No- the move to local state was an accident actually ☹️ I missed this in the review for https://github.com/brave/brave-core/pull/15129

By moving purchase token to local state, we broke Android in 1.45 actually 😱